### DISCUSSION TITLE
Upload 上传大文件progress与后端实际上传进度不一致

### DISCUSSION THREAD

**Discussion by ChenKun1997** [ant-design/ant-design#42983] [Q&A]
上传一个50M的文件，组件给到的progress很快到了100%，后端上传接口却要 pending 20多分钟了，有什么方案可以拿到后端实际的上传进度呢？

**Comment by ByYogi** [comment]
`ChatGPT`
在文件上传过程中，前端和后端的进度可能会有一些差异，导致前端进度很快达到100%，但后端仍在处理文件的情况。这是因为前端的进度是根据文件在客户端的上传情况计算得出的，而后端的进度则取决于服务器端的处理速度和网络传输等因素。

如果你想获取后端实际的上传进度，可以考虑以下几种方案：

1. 询问后端支持：与后端开发人员沟通，了解是否有提供上传进度的接口或方法。有些后端框架或库可能已经提供了获取上传进度的功能，你可以利用它们来获取准确的进度信息。

2. 使用 WebSocket：使用 WebSocket 技术与后端建立实时的双向通信。前端在文件上传过程中，将上传进度实时发送给后端，后端将进度信息推送给前端。这样可以实时获取后端的上传进度，并在前端展示。

3. 分片上传：将大文件拆分成多个小片段进行上传。在每个片段上传完成后，后端可以返回一个片段的上传进度给前端，前端再计算整体的上传进度。这样可以更准确地获取整个文件上传的进度。

4. 轮询后端接口：前端定时向后端发送请求，询问上传进度。后端在接收到请求后，返回上传进度给前端。前端通过轮询这个接口来获取后端实际的上传进度。

需要注意的是，以上方案的可行性和实现方式可能与你所使用的后端技术和框架有关。在选择方案之前，最好与后端开发人员协商并了解后端的具体情况和可支持的方式。
